sqlness/src/interceptor/template.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2024 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use minijinja::Environment;
+use serde_json::Value;
+
+use super::{Interceptor, InterceptorFactory, InterceptorRef};
+
+pub struct TemplateInterceptorFactory;
+
+const PREFIX: &str = "TEMPLATE";
+
+/// Templated query, powered by [minijinja](https://github.com/mitsuhiko/minijinja).
+/// The template syntax can be found [here](https://docs.rs/minijinja/latest/minijinja/syntax/index.html).
+///
+/// Grammar:
+/// ``` text
+/// -- SQLNESS TEMPLATE <json>
+/// ```
+///
+/// `json` define data bindings passed to template, it should be a valid JSON string.
+///
+/// # Example
+/// `.sql` file:
+/// ``` sql
+/// -- SQLNESS TEMPLATE {"name": "test"}
+/// SELECT * FROM table where name = "{{name}}"
+/// ```
+///
+/// `.result` file:
+/// ``` sql
+/// -- SQLNESS TEMPLATE {"name": "test"}
+/// SELECT * FROM table where name = "test";
+/// ```
+///
+/// In order to generate multiple queries, you can use the builtin function
+/// `sql_delimiter()` to insert a delimiter.
```

Comment:
Why we need a special function to insert delimiter, but not write it in literal?

Reply:
```sql
-- SQLNESS TEMPLATE {"aggr": ["sum", "avg", "count"]}
{% for item in aggr %}
SELECT {{item}}(c) from t {%if not loop.last %} {{sql_delimiter()}} {% endif %}
{% endfor %}
;
```

See this demo, `;` is used to separate different SQL, so we can't write it directly in our template.